### PR TITLE
added doc sum react-ui

### PR DESCRIPTION
**Pull request by jaswanth8888**
## Description

Added Doc Sum React UI

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

File: DocSum/docker/xeon/docker_compose.yaml [hunk 1]
```diff
@@ -68,6 +68,23 @@ services:
       - DOC_BASE_URL=${BACKEND_SERVICE_ENDPOINT}
     ipc: host
     restart: always
+  docsum-xeon-react-ui-server:
+    image: opea/docsum--reactui:latest
```
[the remark this answers is not in the document]
Comment (jaswanth8888) [reply DocSum/docker/xeon/docker_compose.yaml:72]: fixed it

**Review by amberjain1 — APPROVED** [review]

File: DocSum/docker/ui/react/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<h1 align="center" id="title">Doc Summary React</h1>
+
+### 📸 Project Screenshots
+
+![project-screenshot](../../../assets/img/docSum_ui_upload.png)
+![project-screenshot](../../../assets/img/docSum_ui_exchange.png)
+![project-screenshot](../../../assets/img/docSum_ui_response.png)
+![project-screenshot](../../../assets/img/docSum_ui_text.png)
```
Comment (yogeshmpandey) [inline DocSum/docker/ui/react/README.md:8]: These do not exist

File: DocSum/docker/xeon/README.md [hunk 3]
```diff
@@ -52,6 +52,22 @@ Then run the command `docker images`, you will have the following Docker Images:
 2. `opea/docsum:latest`
 3. `opea/docsum-ui:latest`
 
+### 4. Build React UI Docker Image
+
+Build the frontend Docker image via below command:
+
+```bash
+cd GenAIExamples/DocSum/docker/ui/
+docker build -t opea/docsum-react-ui:latest --build-arg BACKEND_SERVICE_ENDPOINT=$BACKEND_SERVICE_ENDPOINT -f ./docker/Dockerfile.react .
```
[the remark this answers is not in the document]
Comment (jaswanth8888) [reply DocSum/docker/xeon/README.md:62]: added the command

File: DocSum/docker/ui/react/README.md (hunk 2 above)
Comment (jaswanth8888) [reply DocSum/docker/ui/react/README.md]: fixed it

**Comment by kevinintel** [comment]
tsc: not found

**Comment by kevinintel** [comment]
tsc: not found

File: DocSum/docker/gaudi/README.md [hunk 4]
```diff
@@ -44,12 +44,26 @@ cd GenAIExamples/DocSum/docker/ui/
...
+Build the frontend Docker image via below command:
+
+```bash
+cd GenAIExamples/DocSum/docker/ui/
+export BACKEND_SERVICE_ENDPOINT="http://${host_ip}:8888/v1/docsum"
+docker build -t opea/docsum-react-ui:latest --build-arg BACKEND_SERVICE_ENDPOINT=$BACKEND_SERVICE_ENDPOINT -f ./docker/Dockerfile.react .
+```
+
+Note:
+If docker build fails, update the Base image in `Dockerfile.react` to `node:20.10.0-alpine`
```
Comment (jaswanth8888) [inline DocSum/docker/gaudi/README.md:58]: we observed the same tsc not found issue in gaudi build . so we added this note to update the base image . Please try it with this base image

**Comment by chensuyue** [comment]
Add a new UI service into the test, please also update the test scripts for both xeon and gaudi, 
For Gaudi, add the new docker image build here,  https://github.com/opea-project/GenAIExamples/blob/b4b1d3f1e05ac436a74284d20207a44fbdcda55c/DocSum/tests/test_docsum_on_gaudi.sh#L25
Add `sed -i "s#opea/docsum-react-ui:latest#image: opea/docsum-react-ui:${IMAGE_TAG}#g" docker_compose.yaml` after 
https://github.com/opea-project/GenAIExamples/blob/b4b1d3f1e05ac436a74284d20207a44fbdcda55c/DocSum/tests/test_docsum_on_gaudi.sh#L45

File: DocSum/docker/ui/react/.env
Comment (WenjiaoYue) [inline DocSum/docker/ui/react/.env:1]: In your program, you are using VITE_DOC_SUM_URL, but the variables declared in the .env file are VITE_BACKEND_SERVICE_ENDPOINT and VITE_DATA_PREP_SERVICE_URL.

File: DocSum/docker/ui/react/public/vite.svg
Comment (WenjiaoYue) [inline DocSum/docker/ui/react/public/vite.svg:1]: Do we need to keep vite.svg in our project?

File: DocSum/docker/ui/react/src/assets/react.svg
Comment (WenjiaoYue) [inline DocSum/docker/ui/react/src/assets/react.svg:1]: Do we need to keep react.svg in our project?

File: DocSum/docker/ui/react/src/vite-env.d.ts
Comment (WenjiaoYue) [inline DocSum/docker/ui/react/src/vite-env.d.ts:1]: What is the purpose of keeping this file?

File: DocSum/docker/xeon/docker_compose.yaml
Comment (WenjiaoYue) [inline DocSum/docker/xeon/docker_compose.yaml:1]: Your front_end code does not use DOC_BASE_URL, please rewrite it according to the backend URL you declared.

File: DocSum/docker/ui/react/src/components/DocSum/FileUpload.tsx
Comment (WenjiaoYue) [inline DocSum/docker/ui/react/src/components/DocSum/FileUpload.tsx:1]: Please delete useless code

File: DocSum/docker/gaudi/docker_compose.yaml
Comment (WenjiaoYue) [inline DocSum/docker/gaudi/docker_compose.yaml:1]: Your front_end code does not use DOC_BASE_URL, please rewrite it according to the backend URL you declared

File: DocSum/docker/ui/react/src/components/Shared/Markdown/Markdown.tsx
Comment (WenjiaoYue) [inline DocSum/docker/ui/react/src/components/Shared/Markdown/Markdown.tsx:1]: Please delete useless code

File: DocSum/docker/ui/react/src/vite-env.d.ts
Comment (jaswanth8888) [reply DocSum/docker/ui/react/src/vite-env.d.ts:1]: This file is particular for the editors to recognise the environent variable available

Ref: https://vitejs.dev/guide/env-and-mode.html#intellisense-for-typescript

File: DocSum/docker/ui/react/src/components/DocSum/FileUpload.tsx
Comment (jaswanth8888) [reply DocSum/docker/ui/react/src/components/DocSum/FileUpload.tsx:1]: I have commented out the file type that we would support in future

File: DocSum/docker/ui/react/src/components/Shared/Markdown/Markdown.tsx
Comment (jaswanth8888) [reply DocSum/docker/ui/react/src/components/Shared/Markdown/Markdown.tsx:1]: Removed

File: DocSum/docker/xeon/docker_compose.yaml
Comment (jaswanth8888) [reply DocSum/docker/xeon/docker_compose.yaml:1]: removed DOC_BASE_URL

File: DocSum/docker/gaudi/docker_compose.yaml
Comment (jaswanth8888) [reply DocSum/docker/gaudi/docker_compose.yaml:1]: Removedt he environments as we dont need them

File: DocSum/docker/ui/react/src/assets/react.svg
Comment (jaswanth8888) [reply DocSum/docker/ui/react/src/assets/react.svg:1]: deleted the file

File: DocSum/docker/ui/react/public/vite.svg
Comment (jaswanth8888) [reply DocSum/docker/ui/react/public/vite.svg:1]: deleted the file

File: DocSum/docker/ui/react/.env
Comment (jaswanth8888) [reply DocSum/docker/ui/react/.env:1]: BACKEND_SERVICE_ENDPOINT   is being renamed to VITE_DOC_SUM_URL in dockerfile.react 
```
ARG BACKEND_SERVICE_ENDPOINT
ENV VITE_DOC_SUM_URL=$BACKEND_SERVICE_ENDPOINT
```

File: DocSum/docker/ui/react/src/vite-env.d.ts
Comment (WenjiaoYue) [reply DocSum/docker/ui/react/src/vite-env.d.ts:1]: But the file doesn’t contain the needed content, as everything is commented out. Should it still be kept?

Comment (jaswanth8888) [reply DocSum/docker/ui/react/src/vite-env.d.ts:1]: we have plans to add few variables  so we want to keep it

**Review by yogeshmpandey — APPROVED** [review]

**Comment by chensuyue** [comment]
Please did the same thing for test scripts: https://github.com/opea-project/GenAIExamples/pull/417/files#diff-b51ee89002bc09d93363dcf73ad00bbd681fe027deca914f061d25c6344760b5

File: DocSum/docker/xeon/docker_compose.yaml [hunk 5]
```diff
@@ -68,6 +68,18 @@ services:
       - DOC_BASE_URL=${BACKEND_SERVICE_ENDPOINT}
     ipc: host
     restart: always
+  docsum-xeon-react-ui-server:
+    image: opea/docsum-react-ui:latest
+    container_name: docsum-xeon-react-ui-server
+    depends_on:
+      - docsum-xeon-backend-server
+    build:
```
Comment (chensuyue) [inline DocSum/docker/xeon/docker_compose.yaml:76]: suggest to remove the build part.

File: DocSum/docker/gaudi/docker_compose.yaml [hunk 6]
```diff
@@ -71,6 +71,18 @@ services:
       - DOC_BASE_URL=${BACKEND_SERVICE_ENDPOINT}
     ipc: host
     restart: always
+  docsum-xeon-react-ui-server:
+    image: opea/docsum-react-ui:latest
+    container_name: docsum-gaudi-react-ui-server
+    depends_on:
+      - docsum-gaudi-backend-server
+    build:
```
Comment (chensuyue) [inline DocSum/docker/gaudi/docker_compose.yaml:79]: suggest to remove the build part.

Comment (jaswanth8888) [reply DocSum/docker/gaudi/docker_compose.yaml:79]: build part is need as we cannot inject environmental values to react once the image is built

File: DocSum/docker/ui/react/README.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+<h2>🛠️ Get it Running:</h2>
+
+1. Clone the repo.
+
+2. cd command to the current folder.
+
+3. Modify the required .env variables.
+   ```
+   VITE_DOC_SUM_URL = ''
```
Comment (WenjiaoYue) [inline DocSum/docker/ui/react/README.md:25]: The variable in your .env file is `VITE_BACKEND_SERVICE_ENDPOINT`, so why is it `VITE_DOC_SUM_URL` here?

File: DocSum/docker/gaudi/docker_compose.yaml [hunk 8]
```diff
@@ -71,6 +71,18 @@ services:
...
+      args:
+        - BACKEND_SERVICE_ENDPOINT=${BACKEND_SERVICE_ENDPOINT}
+    ports:
+      - "5174:80"
+    environment:
+      - DOC_BASE_URL=${BACKEND_SERVICE_ENDPOINT}
```
Comment (WenjiaoYue) [inline DocSum/docker/gaudi/docker_compose.yaml:85]: The same issue, you didn't use the DOC_BASE_URL variable in your code. Why did you assign a value to this variable? The variable name assigned in your .env file is DOC_SUM_URL.

File: DocSum/docker/ui/react/.env [hunk 9]
```diff
@@ -0,0 +1 @@
+VITE_BACKEND_SERVICE_ENDPOINT=http://backend_address:8888/v1/docsum
```
Comment (WenjiaoYue) [inline DocSum/docker/ui/react/.env:1]: The variable names in the .env file are not consistent with the ones you are actually using.

File: DocSum/docker/xeon/docker_compose.yaml [hunk 10]
```diff
@@ -68,6 +68,18 @@ services:
...
+      args:
+        - BACKEND_SERVICE_ENDPOINT=${BACKEND_SERVICE_ENDPOINT}
+    ports:
+      - "5174:80"
```
Comment (WenjiaoYue) [inline DocSum/docker/xeon/docker_compose.yaml:80]: In the docker_compose.yaml of Gaudi, you introduced the environment parameter, but Xeon should be the same; why is it not included here?

File: DocSum/docker/ui/react/src/styles/components/_context.scss
Comment (WenjiaoYue) [inline DocSum/docker/ui/react/src/styles/components/_context.scss:1]: Should empty files be deleted?

File: DocSum/docker/ui/react/.env (hunk 9 above)
Comment (jaswanth8888) [reply DocSum/docker/ui/react/.env:1]: renamed it to VITE_DOC_SUM_URL

File: DocSum/docker/gaudi/docker_compose.yaml (hunk 8 above)
Comment (jaswanth8888) [reply DocSum/docker/gaudi/docker_compose.yaml:85]: removed environemnt variables

File: DocSum/docker/ui/react/README.md (hunk 7 above)
Comment (jaswanth8888) [reply DocSum/docker/ui/react/README.md:25]: to run it locally it has to be VITE_DOC_SUM_URL

File: DocSum/docker/xeon/docker_compose.yaml (hunk 10 above)
Comment (jaswanth8888) [reply DocSum/docker/xeon/docker_compose.yaml:80]: removed the environment section in gaudi as it is not needed

**Review by WenjiaoYue — APPROVED** [review]

File: DocSum/tests/test_docsum_on_gaudi.sh [hunk 11]
```diff
@@ -43,6 +44,7 @@ function start_services() {
         echo "using image repository $IMAGE_REPO and image tag $IMAGE_TAG"
         sed -i "s#image: opea/docsum:latest#image: opea/docsum:${IMAGE_TAG}#g" docker_compose.yaml
         sed -i "s#image: opea/docsum-ui:latest#image: opea/docsum-ui:${IMAGE_TAG}#g" docker_compose.yaml
+        sed -i "s#image: opea/docsum-react--ui:latest#image: opea/docsum-react-ui:${IMAGE_TAG}#g" docker_compose.yaml
```
Comment (chensuyue) [inline DocSum/tests/test_docsum_on_gaudi.sh:47]: typo `opea/docsum-react--ui`

Comment (jaswanth8888) [reply DocSum/tests/test_docsum_on_gaudi.sh:47]: fixed it

**Pull request merged**